gpii/node_modules/matchMakerFramework/src/MatchMakerFramework.js

```diff
@@ -81,6 +80,158 @@
             event.fire(matchData, matchMakerPayload);
         });
     };
+	
+    /*
+	 * Returns a string indicating the matchmaker to select from configuration. Depends on the matchmaker payload.
+     */
+    gpii.matchMakerFramework.selectMatchMaker = function (matchMakerPayload) {
+		var result = gpii.matchMakerFramework.GetPreferredMatchMaker(matchMakerPayload);
+		return (result) ? result : gpii.matchMakerFramework.GetCapableMatchMaker(matchMakerPayload);
+    };
+
+    /*
+	 * Returns if a matchmaker payload contains preferences for selecting a matchmaker.
+	 * Returns NULL if no matchmaker preference was found.
+     */
+	gpii.matchMakerFramework.GetPreferredMatchMaker = function (matchMakerPayload) {
+		// Constants
+		var USER_DEFINED_MATCHMAKER_PREF0 = "http://registry.gpii.org/common/matchMakerType";
+		var USER_DEFINED_MATCHMAKER_PREF1 = "http://registry.gpii.net/common/matchMakerType";
+		// Rejection tests
+		if (matchMakerPayload.preferences === undefined || matchMakerPayload.preferences.contexts === undefined || matchMakerPayload.activeContexts === undefined) {
+			return null;
+		}
+		// Search for mm selection preferences
+		var contexts = matchMakerPayload.preferences.contexts;
+		var activeContexts = matchMakerPayload.activeContexts;
+		var foundPriority = -1;
+		var foundMatchMaker;
+		fluid.each(activeContexts, function (activeContext) {
+			if ((activeContext in contexts) && ("preferences" in contexts[activeContext])) {
+				var prefs = contexts[activeContext].preferences;
+				var curPriority = 0;
+				if ("priority" in contexts[activeContext]) {
+					curPriority = contexts[activeContext].priority;
+				}
+				if ((USER_DEFINED_MATCHMAKER_PREF0 in prefs) && (curPriority > foundPriority)) {
+					foundPriority = curPriority;
+					foundMatchMaker = prefs[USER_DEFINED_MATCHMAKER_PREF0];
+				} else if ((USER_DEFINED_MATCHMAKER_PREF1 in prefs) && (curPriority > foundPriority)) {
+					foundPriority = curPriority;
+					foundMatchMaker = prefs[USER_DEFINED_MATCHMAKER_PREF1];
+				}
+			}
+		});
+		return foundMatchMaker;
+    };
+
+    /*
+	 * Returns the Matchmaker that best matches to the playload based on its capabilities.
+	 * See MatchMakerCapabilities.json.
+     */
+	gpii.matchMakerFramework.GetCapableMatchMaker = function (matchMakerPayload) {
+		// Constants
+		var EXCLUDED_MATCHMAKER_PREF0 = "http://registry.gpii.org/common/excludedMatchMakers";
+		var EXCLUDED_MATCHMAKER_PREF1 = "http://registry.gpii.net/common/excludedMatchMakers";
```

Comment:
Duplication here and further down as well... Unless there is a good reason for it, one of them should be removed